src/controllers/bulk/bulksupported.h

```diff
@@ -12,7 +13,8 @@ struct bulk_device_id {
 struct bulk_device_endpoints {
     std::uint8_t in_epaddr;
     std::uint8_t out_epaddr;
-    std::uint8_t interface_number;
+    // we may not know the interface, in which case we should not try to claim it.
+    std::optional<std::uint8_t> interface_number = std::nullopt;
```

Comment:
The initializer is redundant. I would  rather make it explicit in the table, that the value is unknown and we need to fill it to make it work on Windows and macOs.  In addition a brief to-do comment in the table is useful. We may also file a bug as a reminder that these devices are not yet usable with windows.

Reply:
are we sure they only don't work under windows?

Reply:
done

Reply:
I don't know. But it looks like @acolombier as added the claiming to enable the controllers in windows.

Reply:
yeah I saw that. but maybe it also just didn't work under windows because libusb didn't claim the right interface automatically (since from what I could tell an interface needs to be claimed anyways in order to do any operation on it). Do we want to rely on the automatic claiming (if it exists across all backends at all)?

Reply:
It looks like libusb is not able to automatically claim an interface. The auto feature is regarding detaching the kernel driver which is not supported on Windows. 
But we need to test, I have no setup for that.

Reply:
I guessed so because I saw [`auto_claim`](https://github.com/libusb/libusb/blob/467b6a8896daea3d104958bf0887312c5d14d150/libusb/os/windows_winusb.c#L613) in winusb but upon further research its not used anywhere else. I don't have a setup for it either, but I guess we can continue not claiming it until someone with the setup files an issue.

Reply:
Getting libusb running under Windows requires further changes. It is not functional yet, as we need to register a device driver (e.g. the generic WinUSB driver that comes with Windows) for the particular device type. This could be done using https://github.com/pbatard/libwdi but requires elevated user rights.  Either in the Mixxx installer or at runtime of Mixxx itself with temporary elevated rights.
This is why BULK is default off for Windows.